backend/database/versions/9a8dd669a47c_data_source_link.py

```diff
@@ -0,0 +1,61 @@
+"""data_source_link
+
+Revision ID: 9a8dd669a47c
+Revises: bf83d170bb7f
+Create Date: 2021-03-08 16:34:52.801864
+
+"""
+import sqlalchemy as sa
+
+# revision identifiers, used by Alembic.
+from alembic import op
+
+revision = '9a8dd669a47c'
+down_revision = 'bf83d170bb7f'
+branch_labels = None
+depends_on = None
+
+# Enum 'type' for PostgreSQL
+enum_projectlink = "projectlink"
+# Set temporary enum 'type' for PostgreSQL
+tmp_enum_projectlink = "tmp_" + enum_projectlink
+
+# Options for Enum
+old_options = sorted(["DOI","RAW_DATA", "PROTOCOL","LAB_WEBSITE","OTHER"])
+new_options = sorted(["DOI", "RAW_DATA", "PROTOCOL", "LAB_WEBSITE", "OTHER", "DATA_SOURCE"])
+
+# Create enum fields
+old_type = sa.Enum(*old_options, name=enum_projectlink)
+new_type = sa.Enum(*new_options, name=enum_projectlink)
+
+
+def upgrade():
+    # Rename current enum type to tmp_
+    op.execute("ALTER TYPE " + enum_projectlink + " RENAME TO " + tmp_enum_projectlink)
```

Comment:
Is there a reason we cant edit enum options using
```
with op.get_context().autocommit_block():
    op.execute("ALTER TYPE ProjectLinkType ADD VALUE 'DATA_SOURCE'")
    ```

and then for the downgrade delete any links referencing the option and run 
`op.execute("ALTER TYPE ProjectLinkType DROP VALUE 'DATA_SOURCE'")`

instead of renaming/recreating the enum?

Reply:
Older versions of postgres didn't allow easily updating enums, and I knew this way worked.